tests/capture/test_capture_qnode.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for capturing a qnode into jaxpr.
+"""
+from functools import partial
+
+# pylint: disable=protected-access
+import pytest
+
+import pennylane as qml
+from pennylane.capture.capture_qnode import _get_qnode_prim
+
+qnode_prim = _get_qnode_prim()
+
+pytestmark = pytest.mark.jax
+
+jax = pytest.importorskip("jax")
+
+
+@pytest.fixture(autouse=True)
+def enable_disable_plxpr():
+    qml.capture.enable()
+    yield
+    qml.capture.disable()
+
+
+@pytest.mark.parametrize("dev_name", ("default.qubit", "default.qubit.legacy"))
+def test_error_if_shot_vector(dev_name):
+    """Test that a NotImplementedError is raised if a shot vector is provided."""
+
+    dev = qml.device(dev_name, wires=1, shots=(50, 50))
+
+    @qml.qnode(dev)
+    def circuit():
+        return qml.sample()
+
+    with pytest.raises(NotImplementedError):
+        jax.make_jaxpr(circuit)()
+
+
+def test_error_if_no_device_wires():
+    """Test that a NotImplementedError is raised if the device does not provide wires."""
+
+    dev = qml.device("default.qubit")
+
+    @qml.qnode(dev)
+    def circuit():
+        return qml.sample()
+
+    with pytest.raises(NotImplementedError):
+        jax.make_jaxpr(circuit)()
+
+
+@pytest.mark.parametrize("x64_mode", (True, False))
+def test_simple_qnode(x64_mode):
+    """Test capturing a qnode for a simple use."""
+
+    initial_mode = jax.config.jax_enable_x64
+    jax.config.update("jax_enable_x64", x64_mode)
+
+    dev = qml.device("default.qubit", wires=4)
+
+    @qml.qnode(dev)
+    def circuit(x):
+        qml.RX(x, wires=0)
+        return qml.expval(qml.Z(0))
+
+    jaxpr = jax.make_jaxpr(circuit)(0.5)
+
+    eqn0 = jaxpr.eqns[0]
+
```

Comment:
Could define an `exp_dtype = jax.numpy.float64 if x64_mode else jax.numpy.float32` here and reuse it